packages/webviz-core/src/dataProviders/MemoryCacheDataProvider.js

```diff
@@ -393,6 +425,8 @@ export default class MemoryCacheDataProvider implements DataProvider {
       };
       this._resolveFinishedReadRequests();
     }
+    // Implicit return impossible, but satisfy flow:
```

Comment:
In that case you can also throw an error here instead. I think that should also satisfy Flow, and would actually alert us if this assumption was violated.

Reply:
Haha, oh dear -- flow has a problem with the early-return, and eslint has a problem with unreachable code :-). I've made the success case break out of the loop to try to satisfy both. Will rebase it onto the relevant earlier commit later.